2996_reflection/reflection.md

```diff
@@ -3123,6 +3127,22 @@ An expression convertible to a reflection is said to _represent_ the correspondi
 :::
 :::
 
+### [conv.ptr]{.sref} Pointer conversions {-}
+
+Expand paragraph 2 to disallow conversions to `void *` from pointers of consteval-only types.
```

Comment:
Uh, what? Why? This effectively prohibits any type erasure involving consteval-only types, even at compile time.

Reply:
What's the problem we're trying to solve here? Okay, so I have a `void*` to an `info`. What am I going to do with it? 

```cpp
constexpr meta::info x = ^^int;
int main() {
    void* p = (void*)&x;
    // ???
}
```

All operations on this thing are `consteval` so it's not like I can actually call anything, since `p` isn't constant. What else can I do with the thing? `bit_cast` it to `u64` or something?

Reply:
Pretty much. I suppose we can say that the object representation of all `std::meta::info` values is unspecified, and maybe...that's good enough?

Reply:
So I'm currently rewriting P3032 to do immediate-escalating expressions and I think we want to say something like this (terms TBD)

> An expression is consteval-only if it is:
> 
> * an id-expression naming a consteval function,
> * an explicit or implicit call to a consteval function, or
> * has consteval-only type or is an explicit or implicit conversion to **or from** a consteval-only type.

Then something like:

> An expression is *immediate-escalating* if it is a consteval-only expression unless:
> 
> * it is initially in an immediate function context, or
> * it either is a constant expression or is a subexpression of a constant expression.

And then I think the rest of the words kind of work out? 

I'm actually not 100% sure on the "or from" part. But is it really feasible to have a conversion *from* a consteval-only type that *is not* a constant expression/

Reply:
Addressed this with what we reached on Monday's call: consteval-only-erased pointers and references aren't permitted results of constant expressions.